File: hw1/hw1.java
import java.io.*;
import java.util.Scanner;
import java.lang.Math;

public class hw1
{

    public static double seadragon(double a,double b,double c)
    {
        if(a<=0||b<=0||c<=0) return -1;
        double s=a+b+c;
        s/=2;
        return Math.sqrt(s*(s-a)*(s-b)*(s-c));
    }

    public static void main(String argc[]) throws IOException
    {
        System.out.println("triangle area calculater");
        Scanner sc = new Scanner(System.in);
        
        FileWriter fw = new FileWriter("java_result.txt",true);
        while(true)
        {

        
        
        double a,b,c,area;
        
        System.out.println("please input first number");
        a=sc.nextDouble();
        System.out.println("please input second number");
        b=sc.nextDouble();
        System.out.println("please input third number");
        c=sc.nextDouble();
        if(a==-1&&b==-1&&c==-1)
        {
            System.out.println("end of program");
            fw.write("end of program\n");
            fw.close();
            
            break;
        }
        System.out.println("a="+a);
        fw.write("a="+a+"\n");
        
        System.out.println("b="+b);
        fw.write("b="+b+"\n");
        
        System.out.println("c="+c);
        fw.write("c="+c+"\n");
        
        area=seadragon(a,b,c);
        if(!(area>0))
        { 
            System.out.println("Invalid input.Please try again");
            fw.write("Invalid input.Please try again\n");
            
        }
        else 
        {
            System.out.println("area="+area);
            fw.write("area="+area+"\n");
            
        }


    }

    }
}

File: hw1/hw1.py
import math
def seadragon(a,b,c):
    if a<=0 or b<=0 or c<=0:
        return -1
    
    
    s=a+b+c
    big=max(a,b,c)
    if s-big<big:
        return -1

    s=s/2
    return math.sqrt(s*(s-a)*(s-b)*(s-c))



print("triangle area calculater")
f=open("py_result.txt",'a')
while True:
    
    a=input('please input first number')
    a=float(a)
    b=input('please input second number')
    b=float(b)
    c=input('please input third number')
    c=float(c)
    if a==-1 and b==-1 and c==-1:
        print("end of program")
        f.write("end of prorram\n")
        f.close()
        break
    print("a =",a)
    st="a ="+ str(a)+'\n'
    f.write(st)
    print("b =",b)
    st="b ="+ str(b)+'\n'
    f.write(st)
    print("c =",c)
    st="c ="+str(c)+'\n'
    f.write(st)
    area=seadragon(a,b,c)
    if area<=0:
        print("invalid input.please try again")
        f.write("invalid input.please try again\n")
    else:
        print("area=",area)
        st="area ="+str(area)+'\n'
        f.write(st)






